docs/hibernating-clusters.md

```diff
@@ -0,0 +1,121 @@
+# Hibernating Clusters
+
+## Overview
+
+Starting with OpenShift version 4.4.8, clusters can be stopped and started by simply shutting
+down their machines and starting them back up. The only additional requirement to bring a cluster
+back up from a stopped state is that CSRs be approved for the cluster's nodes when certificates
+have expired since the cluster was active.
+
+Hive can automate the process of stopping/starting clusters via its API by allowing the user to
+set a the desired state of the cluster in the ClusterDeployment spec. Both API and controller
+changes are required to support this feature.
+
+## API Changes
+
+The ClusterDeploymentSpec should allow setting whether machines are in a running state or in
+a hibernating state.
+
+```go
+type ClusterDeploymentSpec struct {
+  // ... other fields
+
+  // Hibernating indicates whether a cluster should be hibernating. Default is false.
+  Hibernating  `json:"hibernating,omitempty"`
+}
+```
+
+On the status side, a cluster's state is reflected by the following condition types:
+
+```go
+const (
+  // ... other constants
+
+  // ClusterStartingConditionType is set when the ClusterDeployment is transitioning
+  // from a hibernating state to a running state.
+  ClusterStartingConditionType ClusterDeploymentConditionType = "ClusterStarting"
+
+  // ClusterRunningConditionType is set when the ClusterDeployment is in a running and
+  // ready state.
+  ClusterRunningConditionType ClusterDeploymentConditionType = "ClusterRunning"
```

Comment:
Will this be true for clusters that have never hibernated?

Reply:
Yes, but if not present, we could assume that the cluster is running since that's the default state.